lib/listen/event/loop.rb

```diff
@@ -6,97 +6,86 @@
 module Listen
   module Event
     class Loop
+      include Listen::FSM
+
       class Error < RuntimeError
-        class NotStarted < Error
-        end
+        class ThreadFailedToStart < Error; end
+        class AlreadyStarted < Error; end
```

Comment:
I wonder if this should inherit from `NotStarted` to preserve existing semantics where possible.

Reply:
Yes, understood. In this context... if someone calls start, and it's already started, that's a kind of `NotStarted` because it was already started. If you can think of something better, it's worth considering.

Reply:
Restored to its old behavior: https://github.com/guard/listen/pull/482/commits/0b236c1ab903c8653626e8d6404fd1c680238aa8